jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -540,6 +542,33 @@ private boolean isDDLCommand(int updatedCount, int columnCount) throws SQLExcept
     return queries;
   }
 
+  private void executePreCode(Connection connection) {
+    String precode = getProperty(ZEPPELIN_PRECODE_PROPERTY_KEY);
+    if (StringUtils.isNotEmpty(precode)) {
+      logger.info("Run SQL precode '{}'", precode);
+      try {
+        Statement statement = connection.createStatement();
+        statement.execute(precode);
+        if (!connection.getAutoCommit()) {
+          connection.commit();
+        }
+        if (statement != null) {
+          statement.close();
+        }
+      } catch (SQLException e) {
+        logger.error("Cannot create precode statement", e);
```

Comment:
Yes, I want the main code will run anyway. Although it may be worth it to generate an error